core/src/main/scala/kafka/server/BrokerToControllerChannelManagerImpl.scala

```diff
@@ -208,7 +209,9 @@ class BrokerToControllerRequestThread(networkClient: KafkaClient,
   }
 
   private[server] def handleResponse(request: BrokerToControllerQueueItem)(response: ClientResponse): Unit = {
-    if (response.wasDisconnected()) {
+    if (isTimedOut(response)) {
+      debug(s"The request failed to send to the controller after timeout ${clientRequestTimeout} ms: $response")
```

Comment:
Do we need to call the request callback here so we can propagate the error up to the client?

Reply:
I think the `failExpiredRequests` in the InterBrokerSendThread level shall be fired and reach the callback.